### ISSUE TITLE
Clock synchronizer extension fails to connect to board in 0.4.0-build230821 AEON3

### ISSUE THREAD

**Issue by RoboDoig** [SainsburyWellcomeCentre/aeon_acquisition#141]
Running the clock synchronizer extension gives runtime error: the device ID 1158 on <COM> was unexpected.

**Comment by glopesdev** [comment]
@RoboDoig can you check if the board is a synchroniser, or a timestamp generator gen3? I had the same problem and had to provide bindings for both devices, but which one needs to be used will be rig-specific.

**Comment by glopesdev** [comment]
Btw, for reference you can quickly check all device IDs in: https://github.com/harp-tech/protocol/blob/main/whoami.md

Just did a quick check and it looks like it is indeed a timestamp generator gen3, so you need to use the TimestampGenerator extension instead of the ClockSynchroniser.

**Comment by RoboDoig** [comment]
Closing this issue, had to update the Synchronizer extension to use the timestamp generator gen3 as you said.

My confusion came from how the Harp device node updates its identity in the IncludeWorkflow. As it was previously saved as a synchronizer device there was a Harp ID mismatch. Once I manually reconnected to the COM port before running the workflow the identity updated as a timestamp generator.
